control-plane/agents/src/bin/core/volume/snapshot_operations.rs

```diff
@@ -252,10 +253,20 @@ impl ResourcePruning for OperationGuardArc<VolumeSnapshot> {
         registry: &Registry,
         max_prune_limit: Option<usize>,
     ) -> Result<(), SvcError> {
-        // If the transaction is the only transaction or there are no transactions, then no cleanup
-        // is needed. Also if no nodes are online don't attempt.
-        if self.as_ref().metadata().transactions().len() <= 1
-            || !self.check_nodes_availability(registry).await
+        // Check if pruning is allowed based on status and transactions count. Also if no nodes are
+        // online don't attempt.
+        fn pruning_allowed(status: &VolumeSnapshotSpecStatus, transactions: usize) -> bool {
+            match status {
+                VolumeSnapshotSpecStatus::Creating => transactions > 0,
+                VolumeSnapshotSpecStatus::Created(_) => transactions > 1,
+                _ => false,
+            }
+        }
+
+        if !pruning_allowed(
+            self.as_ref().status(),
+            self.as_ref().metadata().transactions().len(),
+        ) || !self.check_nodes_availability(registry).await
```

Comment:
why not simply assign a variable here? example:

```suggestion
        let transactions = self.as_ref().metadata().transactions().len();
        let pruning_allowed = match self.as_ref().status() {
            VolumeSnapshotSpecStatus::Creating => transactions > 0,
            VolumeSnapshotSpecStatus::Created(_) => transactions > 1,
            _ => false,
        };

        if !pruning_allowed || !self.check_nodes_availability(registry).await
```

Reply:
Yeah, we can.